libminifi/include/core/logging/LoggerConfiguration.h

```diff
@@ -98,15 +98,15 @@ class LoggerConfiguration {
    */
   void initialize(const std::shared_ptr<LoggerProperties> &logger_properties);
 
-  static std::unique_ptr<io::InputStream> getCompressedLog(bool flush = false) {
-    return getCompressedLog(std::chrono::milliseconds{0}, flush);
+  static std::vector<std::unique_ptr<io::InputStream>> getCompressedLogs(bool flush = false) {
+    return getCompressedLogs(std::chrono::milliseconds{0}, flush);
```

Comment:
I think it would be better to create a new stream class that delegates to a list of inner streams. What do you think?

Reply:
I think it's a good idea, unfortunately I tried doing it and it cannot be handled as a single stream because of the compression. As the segments are compressed individually when forwarding the input stream to the zlib decompressor it only returns the first segment decompressed, probably due to finding the end of the archive.

Reply:
This new stream class that handles delegating to an inner list of streams could handle reading from the next stream when the first one is over, and hide the complexity inside the new class. Zlib could be involved in a lower level, like this:
ConcatStream contains std::vector<InputStream>, where each stream is a ZlibDecompressStream
Each ZlibDecompressStream contains the original BufferStream (or whatever) with the compressed data.

Reply:
Yes, that was the approach I tried in a previous commit, but the logger test with multiple segments failed with this implementation. https://github.com/lordgamez/nifi-minifi-cpp/commit/2f3d2c0800f5b9fc29e3f3fde64dbcd8d9ede283

I tried modifying the `ZlibDecompressStream` class to be able to handle this kind of `InputStreamBundle` but evenn if I try to adjust the zlib libraries `z_stream` input and output pointers the decompression after the first segment still fails to read the proper data: https://github.com/lordgamez/nifi-minifi-cpp/commit/7ac83b33051fe672f984d0885c198b3baa07e1d6

Reply:
I was hoping for some code simplification, but since it's not trivial, I don't want to block this PR with it.